docs/hacking-an-agent.md

```diff
@@ -1,99 +1,222 @@
 ## Preliminaries
 
-These instructions detail the Python code you need for running an AEA outside the `cli` tool.
+These instructions detail the Python code you need for running an AEA outside the `cli` tool, using the code interface. 
 
-<div class="admonition note">
-  <p class="admonition-title">Note</p>
-  <p>You have already coded up your agent. See the <a href="../aea/skill-guide/" target=_blank>build your own skill guide</a> for a reminder.</p>
-</div>
+  
+This guide assumes you have already followed the [quick start](quickstart.md) guide and so have the framework installed and the packages and scripts directory downloaded into the directory you are working in.
 
 
+## Create private key
+Before we make our agent we make a private key to be used in the wallet. Open a terminal and type:
+    
+``` bash
+python scripts/generate_private_key.py my_key.txt
+```
+
 ## Imports
 
+Now we can start to write our python script. 
+
 First, import the necessary common Python libraries and classes.
 
 ``` python
 import os
 import time
 from threading import Thread
+import yaml
 ```
 
 Then, import the application specific libraries.
 
 ``` python
-from aea.aea import AEA 
+from aea.aea import AEA
+from aea import AEA_DIR
+from aea.skills.base import Skill
 from aea.connections.stub.connection import StubConnection
 from aea.crypto.ledger_apis import LedgerApis
 from aea.crypto.wallet import Wallet
-from aea.mail.base import Envelope
-from aea.protocols.default.message import DefaultMessage
 from aea.protocols.default.serialization import DefaultSerializer
 from aea.registries.base import Resources
+from aea.configurations.base import ProtocolConfig
+from aea.protocols.base import Protocol
 ```
 
+Set up a variable pointing to where the packages directory is located - this should be our current directory. 
+``` python
+root_dir = "./"
+```
 
+## Clearing the input and output files
+We will use the stub connection to pass envelopes in and out of the agent. Ensure that any input and output text files are removed before we start.
+``` python
+# Ensure the input and output files do not exist initially
+input_filename = "input.txt"
```

Comment:
maybe it is easier to do that via command line, like `rm input.txt && touch input.txt` etc?

Reply:
Yes, this might be better though I would like to leave this as is so I can wrap this up (your proposed change requires, me to do code changes and re-testing for development and stranded installs etc and I think the current setup is basically fine).